# Project 1 Report

### Question 1

Overview: How many rows (samples) and columns (features) are present in the dataset?

In [5]:
import pandas as pd
df=pd.read_csv('Project1-Classification.csv')
df.shape

(3150, 8)

##### There are 3150 rows (samples) and 8 columns (features) in the data set. 

Interpret Plots: Provide qualitative interpretations of the histograms.

##### The alpha-numeric histogram right skewed with an average around 2,500 characters. Additionally, most alpha-numeric character counts lie within 0 and 10,000.
##### The leaf_label histogram is evenly distributed as all categories have the same frequency, which is 350.
##### The root_label histogram is not evenly distributed. The sports category appears more frequently than the climate category; sports appears 1750 times while climate appears 1400 times. 

### Question 2

Report the number of training and testing samples

##### There are 2520 training samples and 630 testing samples

### Question 3

What are the pros and cons of lemmatization versus stemming? How do these processes affect
the dictionary size?
• min df means minimum document frequency. How does varying min df change the TF-IDF
matrix?
• Should I remove stopwords before or after lemmatizing? Should I remove punctuations before
or after lemmatizing? Should I remove numbers before or after lemmatizing? Hint: Recall
that the full sentence is input into the Lemmatizer and the lemmatizer is tagging the position
of every word based on the sentence structure.
• Report the shape of the TF-IDF-processed train and test matrices. The number of rows should
match the results of Question 2. The number of columns should roughly be in the order of
k×103
. This dimension will vary depending on your exact method of cleaning and lemmatizing
and that is okay.
